container/containerd/handler.go

```diff
@@ -164,22 +172,33 @@ func (h *containerdContainerHandler) ContainerReference() (info.ContainerReferen
 	return h.reference, nil
 }
 
-func (h *containerdContainerHandler) needNet() bool {
+func maybeRemoveNet(metrics container.MetricSet, labels map[string]string) container.MetricSet {
 	// Since containerd does not handle networking ideally we need to return based
 	// on includedMetrics list. Here the assumption is the presence of cri-containerd
-	// label
-	if h.includedMetrics.Has(container.NetworkUsageMetrics) {
-		//TODO change it to exported cri-containerd constants
-		return h.labels["io.cri-containerd.kind"] == "sandbox"
+	// label.
+
+	// TODO change it to exported cri-containerd constants.
+	if labels["io.cri-containerd.kind"] == "sandbox" {
+		return metrics
 	}
-	return false
+
+	// This is not a pod/sandbox container; remove network metrics.
+
+	// Check if there is anything we can remove.
+	if !metrics.HasAny(container.AllNetworkMetrics) {
+		return metrics
+	}
+
```

Comment:
Is there any reason we need to first check if the network metrics are included and only if true, then we call `metrics.Difference`? Shouldn't it be safe to always call `metric.Difference` (if the network metrics are not included, it should be a no-op)?

Or is this check more a performance optimization to avoid creating a new `MetricSet`?

Reply:
Yes, this is a performance optimization -- why bothering with copying if we don't have to? The downside is code looks a bit less compact than it could be -- but still pretty readable.

Reply:
> Could you provide tests for `maybeRemoveNet()` to avoid weird regression in future, please?

Done